MonoZelda/Items/ItemFactory.cs

```diff
@@ -1,27 +1,31 @@
 ﻿using Microsoft.Xna.Framework.Graphics;
-using MonoZelda.Scenes;
-using MonoZelda.Collision;
+using Microsoft.Xna.Framework;
+using MonoZelda.Controllers;
+using Microsoft.Xna.Framework.Content;
+using MonoZelda.Sprites;
 using MonoZelda.Items.ItemClasses;
 using System;
-using MonoZelda.Controllers;
 
 namespace MonoZelda.Items;
 
 public class ItemFactory
-{
-    private GraphicsDevice graphicsDevice;
+{ 
     private CollisionController collisionController;
+    private ContentManager contentManager;
 
-    public ItemFactory(GraphicsDevice graphicsDevice, CollisionController collisionController)
+    public ItemFactory( CollisionController collisionController, ContentManager contentManager)
     {
-        this.graphicsDevice = graphicsDevice;
         this.collisionController = collisionController;
+        this.contentManager = contentManager;  
     }
 
-    public T CreateItem<T>() where T : IItem
+    public void CreateItem(ItemList itemName, Point spawnPosition)
     {
-        // Create an instance of the item, passing in the common objects to the constructor
-        return (T)Activator.CreateInstance(typeof(T), collisionController, graphicsDevice);
+        var itemDict = new SpriteDict(contentManager.Load<Texture2D>("Sprites/items"), SpriteCSVData.Items, 0, new Point(0, 0));
+        var itemType = Type.GetType($"MonoZelda.Items.ItemClasses.{itemName}");
+        IItem item = (IItem)Activator.CreateInstance(itemType);
+        item.itemSpawn(itemDict, spawnPosition, collisionController);
+
```

Comment:
Is this code what connects the item factory to the level loading? Will items now spawn for each room when this PR is merged?

Reply:
Yes, this method is called in the DungeonScene class, where the items from the ItemSpawn.GetItemSpawn() list are iterated over and createItem is called on each of them. Updating the Google sheet will items will automically spawn items because of how room loading has been implemented